app.py

```diff
@@ -31,46 +30,158 @@ def init_app(self):
+            parent_dir = os.path.dirname(path)
+            self._launch_filemanager_for_folder(parent_dir)
+
+            
+    def _launch_filemanager_for_folder(self, path):
+        """
+        This method will take a path to a folder and open it in the OS's default file manager.
+        :param path: A folder path
+        :raises: Exception: if the Platform is not supported
+        :raises: ValueError: if the path is not a valid directory.
+        """
+        self.log_debug("Launching file system viewer for folder %s" % path)
+
+        # Check that we don't have a file path.
+        if not os.path.isdir(path):
+            raise ValueError(
+                "The path \"%s\" is not a valid directory." % path)
+
+        # get the setting
         system = sys.platform
-        
+
         # run the app
-        if system == "linux2":
+        if system.startswith("linux2"):
             cmd = 'xdg-open "%s"' % path
         elif system == "darwin":
             cmd = 'open "%s"' % path
         elif system == "win32":
             cmd = 'cmd.exe /C start "Folder" "%s"' % path
         else:
             raise Exception("Platform '%s' is not supported." % system)
-        
+
         self.log_debug("Executing command '%s'" % cmd)
         exit_code = os.system(cmd)
         if exit_code != 0:
             self.log_error("Failed to launch '%s'!" % cmd)
 
-    
+    def _launch_filemanager_for_file(self, path):
+        """
+        This method will take a path to a file and open it in the OS's default file manager.
+        This is only used for files, for folders use _launch_filemanager_for_folder.
+        :param path: A file path
+        :raises: Exception: if the Platform is not supported
+        :raises: ValueError: if the path is not a valid file.
+        """
+        self.log_debug("Launching file system viewer for file %s" % path)
+
+        if not os.path.isfile(path):
+            raise ValueError(
+                "The path \"%s\" is not a valid file path." % path)
+
+        # get the setting
+        system = sys.platform
+
+        # build a command that will open and ideally select the file in the OS's default file manager.
+        if system.startswith("linux"):
+            # Can't find a reliable way to open a file browser and select a file on linux,
+            # so if we get passed a file path, only open up the folder.
+            cmd = 'xdg-open "%s"' % os.path.dirname(path)
+        elif system == "darwin":
+            # -R causes the open command to open a finder window and select the file within the parent directory.
+            cmd = 'open -R "%s"' % path
+        elif system == "win32":
+            # /select makes windows select the file within the explorer window
+            cmd = 'explorer /select,"%s"' % path
+        else:
+            raise Exception("Platform '%s' is not supported." % system)
+
+        self.log_debug("Executing command '%s'" % cmd)
+        exit_code = os.system(cmd)
```

Comment:
I would feel much more comfortable if you instead passing in a array of arguments and using the subprocess module I can imagine how someone might be able to inject stuff inside your code by having " in the file name.

Reply:
Hey JF when you mean to pass an array of arguments and use the subprocess module, you mean instead of building a cmd string, pass each section of the cmd as a list of string to the subprocess module? 
The `_launch_filemanager_for_file` method its self would still be passed a string path though? So couldn't stuff still be injected into the file path name?

Reply:
Exactly. it would look something like `subprocess.call(["explorer", "/select,", path])`

If you really can't use subprocess, you need to consider using [pipes.quote](https://docs.python.org/2/library/pipes.html?highlight=quote#pipes.quote) which will escape the path properly for shell usage.